Tensile/SolutionStructs.py

```diff
@@ -1814,9 +1814,13 @@ def setGlobalLoadTileDimClassic(state, tc, numLoads, totalVectorsCoalesced, tota
           // state["NumLoadsCoalesced%s"%tc]
       state["LSP%s"%tc] = int(math.ceil(float(state["DepthU"]) / state["NumLoadsPerpendicular%s"%tc]))
     else:
-      state["LSC%s"%tc] = int(math.ceil(float(state["DepthU"]) / state["NumLoadsCoalesced%s"%tc]))
-      state["LSP%s"%tc] = state["MacroTile%s"%tc] \
-          // state["NumLoadsPerpendicular%s"%tc]
+      if state["TransposeLDS"] == 1:
+        state["LSC%s"%tc] = int(math.ceil(float(state["DepthU"]) / state["NumLoadsCoalesced%s"%tc]))
+        state["LSP%s"%tc] = min(state["MacroTile%s"%tc], ((state["NumThreads"]*state["GlobalReadVectorWidth"])//state["DepthU"]))
```

Comment:
seems transpose would need to change both LSC and LSP ?

Reply:
fixed.. do not need to change for TRansposeLDS=1